src/impy/kinematics.py

```diff
@@ -16,8 +16,71 @@
 """
 
 import numpy as np
-from impy import pdata
+from impy import pdata, impy_config
 from impy.util import info
+import abc
+import particle
+
+
+def _is_AZ_tuple(arg):
+    if isinstance(arg, tuple):
+        if len(arg) != 2:
+            raise ValueError(
+                "tuple (A, Z) should have len == 2, but given = {0}".format(arg)
+            )
+        if not isinstance(arg[0], int):
+            raise ValueError(
+                "1st entry of (A, Z) should be 'int' type, but it is {0} = {1}".format(
+                    type(arg[0]), arg[0]
+                )
+            )
+        if not isinstance(arg[1], int):
+            raise ValueError(
+                "2nd entry of (A, Z) should be 'int' type, but it is {0} = {1}".format(
+                    type(arg[1]), arg[1]
+                )
+            )
+        return True
+    else:
+        return False
+
+
+class _FromParticleName:
+    all_pdgs = {p.name: int(p.pdgid) for p in particle.Particle.findall()}
+    all_pdgs.update(
+        {p.programmatic_name: int(p.pdgid) for p in particle.Particle.findall()}
+    )
+    all_pdgs.update(
+        photon=22,
+        Higgs=25,
+        proton=2212,
+        antiproton=-2212,
+        neutron=2112,
+        antineutron=-2112,
+    )
+
+    @staticmethod
+    def _get_pdg(pname):
+        try:
+            return _FromParticleName.all_pdgs[pname]
+        except KeyError:
+            raise ValueError("Particle with name = {0} is not found".format(pname))
+
+    @staticmethod
+    def _get_AZ(arg):
```

Comment:
No change necessary, because it is not important, but since the whole class is already marked as private, you don't have to also mark its methods as private.

Reply:
I see. Nothing changed